### PR TITLE
Changes for EOS-21953 CORTX-HA: Migrate to new IEM framework for sending an IEM

### PR DESCRIPTION
**Pull request by ArchanaLimaye**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
   EOS-21953  Need to create IEM using new framework
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
    No
  </code>
</pre>
## Problem Description
<pre>
  <code>
     Need to create IEM using new framework
  </code>
</pre>
## Solution
<pre>
  <code>
    Created the IEM using new framework provided by cortx utils
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
   Called the function create IEMs from single node dev deployment setup.
   Called the function generate_iem and confirmed that IEM is created 
   Also confrimed that the IEM sent on message bus using the new IEM framework.
  This confirmtation was done by using a dummy receiver for message_type IEM

 For node DOWN event on srvnode-1:
 - Called the function : generate_iem('srvnode-1', 'node', 'lost')
 -  following IEM was received by the dummy receiver 
 
 {'iem': {'version': '1', 'info': {'severity': 'warning', 'type': 'Software', 'event_time': 1629720630.45687}, 'location': {'cluster_id': 'e766bd52-c19c-45b6-9c91-663fd8203c2e', 'site_id': '1', 'rack_id': '1', 'node_id': '0', 'host': 'srvnode-1'}, 'source': {'site_id': 1, 'rack_id': 1, 'node_id': 0, 'component': 'HA', 'module': 'Node'}, 'contents': {'event': '0001', 'message': 'The cluster has lost srvnode-1 server. System is running in degraded mode. For more information refer the Troubleshooting guide. Extra Info: host=srvnode-1; status=lost;'}}}

For node UP event for srvnode-1:
 - Called the function : generate_iem('srvnode-1', 'node', 'member')
- Following IEM was received by the dummy receiver 

 {'iem': {'version': '1', 'info': {'severity': 'informational', 'type': 'Software', 'event_time': 1629720675.9106133}, 'location': {'cluster_id': 'e766bd52-c19c-45b6-9c91-663fd8203c2e', 'site_id': '1', 'rack_id': '1', 'node_id': '0', 'host': 'srvnode-1'}, 'source': {'site_id': 1, 'rack_id': 1, 'node_id': 0, 'component': 'HA', 'module': 'Node'}, 'contents': {'event': '0002', 'message': 'Server srvnode-1 has joined back the cluster. System is restored. Extra Info: host=srvnode-1; status=member;'}}}

 
  </code>
</pre>

### CODE REVIEW

File: ha/core/system_health/get_system_attributes.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import ast
+
+from ha  import const
+from ha.core.system_health.system_health import SystemHealth, SystemHealthManager
+from ha.core.config.config_manager import ConfigManager
+from ha.core.error import ClusterManagerError
+
+class NodeParams:
+    @staticmethod
+    def get_node_map(node_name: str) -> dict:
```
Comment (ajay-paratmandali) [inline ha/core/system_health/get_system_attributes.py:28]: get_system_attributes.py

Why this class needed
separate work
can we combine it in other class

Comment (ArchanaLimaye) [reply ha/core/system_health/get_system_attributes.py:28]: We need to get the node map in multiple places in our code.
So separate function is created.
There are also other data points that are needed : ex:  get node_id given a node_name or vice -versa.

All those can be added in this file . So created separate file .

In future there can be parameters specific to other entities like enclosure / storageset etc. So class is created

File: ha/alert/iem.py [hunk 2]
```diff
@@ -37,51 +41,66 @@ def __init__(self):
...
+        component = module_type.get('component')
+        module = module_type.get('module')
+        event_id = module_type.get('event').get(event_type).get('ID')
+        desc = module_type.get('event').get(event_type).get('desc')
+        description = re.sub("\$host", node, desc)
+        description = re.sub("\$status", event_type, description)
+        Log.info(f'Sending an IEM for: {component}-{module} to IEM framework. Description:  {description}')
+
+        try:
+            EventMessage.init(component, source)
```
Comment (Madhura-08) [inline ha/alert/iem.py:85]: Can we initialize this in the constructor of this class?

File: ha/core/system_health/get_system_attributes.py (hunk 1 above)
Comment (ajay-paratmandali) [reply ha/core/system_health/get_system_attributes.py:28]: we can also access node map with system health

key = self._prepare_key(const.COMPONENTS.NODE_MAP.value, node_id=node_id)

File: ha/alert/iem.py (hunk 2 above)
Comment (ajay-paratmandali) [reply ha/alert/iem.py:85]: if it is static method then may not need it 
but goot to have in __init__

File: ha/core/system_health/get_system_attributes.py (hunk 1 above)
Comment (ArchanaLimaye) [reply ha/core/system_health/get_system_attributes.py]: Moved the function to system_health.py

File: ha/alert/iem.py (hunk 2 above)
Comment (ArchanaLimaye) [reply ha/alert/iem.py]: Moved to init

**Review by ajay-paratmandali — APPROVED** [review]

**Review by Madhura-08 — APPROVED** [review]

**Pull request merged**